packages/api/src/controllers/taskController.js

```diff
@@ -39,6 +39,51 @@ const adminRoles = [1, 2, 5];
 //   return false;
 // };
 
+async function getTaskAssigneeAndFinalWage(farm_id, user_id, task_id) {
+  const {
+    assignee_user_id,
+    assignee_role_id,
+    wage_at_moment,
+    override_hourly_wage,
+  } = await TaskModel.getTaskAssignee(task_id);
+  const { role_id } = await UserFarmModel.getUserRoleId(user_id);
+  if (!canCompleteTask(assignee_user_id, assignee_role_id, user_id, role_id)) {
+    throw new Error("Not authorized to complete other people's task");
+  }
+
+  const finalWage = { wage_at_moment: override_hourly_wage ? wage_at_moment : null };
```

Comment:
I think if we still called this one `wagePatchData` it'd be more clear to me that this is an object containing the patch data to be sent out rather than an actual wage amount

Reply:
I wanted to make it clear that this function is for completing task (I could have added a comment) and decided to name the function "...FinalWage" and had the variable name aligned with the function name.
I guess if I move the code to the model, that would not be a concern?
I'm not sure about the best practices, so I will wait for Joyce's response!